lib/Monitor/Staging.pm

```diff
@@ -68,55 +67,56 @@ sub find_live {
-                  # check the id_run corresponds to a run with a status of "run pending" to reduce the
-                  # chance of incorrect updates if an incorrect id_run is entered by the loaders e.g. missing
-                  # digit so that it matches an old run which is "run canceled" such that the automatic run
-                  # folder deletion process would start removing it...
-                  # If a run is canceled (or run status changed) before this, the runfolder may not be processed
-                  # automatically...
-                  my $run_status = $run_row->current_run_status_description();
-                  if ( $run_status ne qq[run pending] ) {
-                      warn "Skipping $run_dir - the id_run $id_run may be wrong" .
-                           " as this run has a status of $run_status\n";
-                      next;
-                  }
-                  # check the instrument name is what is expected for this run
-                  my $db_external_name = $run_row->instrument->external_name();
-                  if (! $db_external_name ){
-                      warn qq[No instrument external_name for '$id_run' found in database, skipping\n];
-                      next;
-                  }
-                  my $staging_instrument_name;
-                  try {
-                      $staging_instrument_name = $check->instrument_name(); # from RunInfo.xml
-                      warn "Retrieved instrument_name $staging_instrument_name\n";
-                  } catch {
-                      warn "error retrieving instrument_name from RunInfo.xml\n";
-                  };
-                  if ( ! defined $staging_instrument_name ) {
-                      warn "instrument_name is undefined, skipping $run_dir\n";
-                      next;
-                  } else {
-                      if ($db_external_name ne $staging_instrument_name) {
-                          warn "Skipping $run_dir - instrument name mismatch" .
-                             " for run $id_run, '$staging_instrument_name' on staging," .
-                             " '$db_external_name' in the database.\n";
-                          next;
-                      }
-                      $run_row->update({'folder_name' => $run_folder}); # or validation will fail
-                  }
-                }    
-                if ( npg_tracking::illumina::run::folder::validation->new(
-                         run_folder          => $run_folder,
-                         id_run              => $id_run,
-                         npg_tracking_schema => $self->schema )->check() ) {
-                    $path_for{$id_run} = $run_dir;
-                    warn "Cached $run_dir for run $id_run\n";
+                my $db_folder_name = $run_row->folder_name;
+                if ( $db_folder_name ) {
+                    if ($db_folder_name ne $run_folder) {
```

Comment:
Given we get the id_run from the runfolder_path are we ever likely to see a mismatch on the folder_name ? I guess there's no harm in checking anyway

Reply:
This is what the deleted role was doing. DJ asked to salvage anythign usable from it. You can imagine a far-fetched scenario when somebody accidentally edited the record.